x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/elasticsearch_assets/component_template.ts

```diff
@@ -10,18 +10,24 @@ import {
   EngineComponentResourceEnum,
   type EngineComponentStatus,
 } from '../../../../../common/api/entity_analytics';
-import type { UnitedEntityDefinition } from '../united_entity_definitions';
+import type { EntityEngineInstallationDescriptor } from '../installation/types';
 
 const getComponentTemplateName = (definitionId: string) => `${definitionId}-latest@platform`;
 
 interface Options {
-  unitedDefinition: UnitedEntityDefinition;
+  /**
+   * The entity engine description id
+   **/
+  id: string;
   esClient: ElasticsearchClient;
 }
 
-export const createEntityIndexComponentTemplate = ({ unitedDefinition, esClient }: Options) => {
-  const { entityManagerDefinition, indexMappings } = unitedDefinition;
-  const name = getComponentTemplateName(entityManagerDefinition.id);
+export const createEntityIndexComponentTemplate = (
+  description: EntityEngineInstallationDescriptor,
+  esClient: ElasticsearchClient
+) => {
+  const { id = '', indexMappings } = description;
```

Comment:
Could this default be dangerous? Should we not require id?

Reply:
[Resolved](https://github.com/elastic/kibana/pull/202888/commits/a75679f2a699c8bf580411f48b1573f50b463972)